src/units/unit.cpp

```diff
@@ -1096,8 +1096,13 @@ color_t unit::xp_color() const
 	const bool far_advance  = static_cast<int>(experience_to_advance()) <= game_config::kill_experience*3;
 
 	color_t color = normal_color;
-
-	if(advances_to().size()){
+	bool major_amla = false;
+	for(const config& adv:get_modification_advances()){
+		if(adv["major_amla"].to_bool ()){
+major_amla=true;
+		}
+	}
+	if(advances_to().size() || major_amla == true){
```

Comment:
This condition is a little weird. The `== true` is redundant, and for this sort of case I think it makes more sense to use `empty()` rather than `size()`.